src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,300 @@
+import {
+  useCreatePaySchedule,
+  useGetAllPaySchedules,
+  useGetPaySchedulePreview,
+  useUpdatePaySchedule,
+} from '@/api/queries/payschedule'
+import { Form } from 'react-aria-components'
+import { useI18n } from '@/i18n'
+import { CalendarDate, parseDate } from '@internationalized/date'
+import { componentEvents } from '@/shared/constants'
+import { Actions, Edit, Head, List } from './_parts'
+
+type MODE = 'LIST_PAY_SCHEDULES' | 'ADD_PAY_SCHEDULE' | 'EDIT_PAY_SCHEDULE' | 'PREVIEW_PAY_SCHEDULE'
+
+type PaySchedulePreviewDraft =
+  operations['get-v1-companies-company_id-pay_schedules-preview']['parameters']['query']
+
+type PayScheduleContextType = {
+  companyId: string
+  handleAdd: () => void
+  handleEdit: (schedule: Schemas['Pay-Schedule']) => void
+  handleCancel: () => void
+  mode: MODE
+  paySchedules: Schemas['Pay-Schedule'][] | null
+  currentPaySchedule:
+    | (Schemas['Pay-Schedule'] & {
+        version?: string
+      })
+    | null
+  payPeriodPreview?: {
+    check_date?: string
+    end_date?: string
+    start_date?: string
+    run_payroll_by?: string
+  }[]
+
+  payPreviewLoading?: boolean
+}
+
+const PayScheduleSchema = v.object({
+  frequency: v.union([
+    v.literal('Every week'),
+    v.literal('Every other week'),
+    v.literal('Twice per month'),
+    v.literal('Monthly'),
+    v.literal('Quarterly'),
+    v.literal('Annually'),
+  ]),
+  anchor_pay_date: v.optional(
+    v.pipe(
+      v.instance(CalendarDate),
+      v.transform(input => input.toString()),
+    ),
+  ),
+  anchor_end_of_pay_period: v.optional(
+    v.pipe(
+      v.instance(CalendarDate),
+      v.transform(input => input.toString()),
+    ),
+  ),
+  day_1: v.optional(v.pipe(v.number(), v.minValue(1), v.maxValue(31))),
+  day_2: v.optional(v.pipe(v.number(), v.minValue(1), v.maxValue(31))),
+  custom_name: v.optional(v.string()),
+  auto_pilot: v.optional(v.string()),
+  custom_twice_per_month: v.optional(v.string()),
+})
+
+export type PayScheduleInputs = v.InferInput<typeof PayScheduleSchema>
+export type PayScheduleOutputs = v.InferOutput<typeof PayScheduleSchema>
+
+export type PayScheduleDefaultValues = RequireAtLeastOne<
+  Partial<
+    Pick<
+      Schemas['Pay-Schedule'],
+      | 'anchor_pay_date'
+      | 'anchor_end_of_pay_period'
+      | 'day_1'
+      | 'day_2'
+      | 'custom_name'
+      | 'auto_pilot'
+    >
+  > & {
+    frequency: 'Every week' | 'Every other week' | 'Twice per month' | 'Monthly'
+  }
+>
+
+const [usePaySchedule, PayScheduleProvider] =
+  createCompoundContext<PayScheduleContextType>('PayScheduleContext')
+export { usePaySchedule }
+
+interface PayScheduleProps extends CommonComponentInterface {
+  companyId: string
+  defaultValues?: PayScheduleDefaultValues
+}
+
+export const PaySchedule = (props: PayScheduleProps & BaseComponentInterface) => {
+  useI18n('Company.PaySchedule')
+  return (
+    <BaseComponent {...props}>
+      <Root {...props}>{props.children}</Root>
+    </BaseComponent>
+  )
+}
```

Comment:
Nit: I think we need to destructure or pick the PayScheduleProps here and pass them explicitly to root, and then the rest can go to the BaseComponent as BaseComponent props. Not a huge deal as it is currently, just ends up passing some extra props to BaseComponent that it doesn't use